server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -300,6 +303,24 @@ public int compare(Interval o1, Interval o2)
     return metrics;
   }
 
+  @GET
+  @Path("/{dataSourceName}/candidates/intervals/{intervals}/numCandidates/{numCandidates}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Iterable<LocatedSegmentDescriptor> getQueryTargets(
+      @PathParam("dataSourceName") String datasource,
+      @PathParam("intervals") String intervals,
+      @PathParam("numCandidates") String numCandidates,
+      @Context final HttpServletRequest req
+  ) throws IOException
+  {
+    List<Interval> intervalList = Lists.newArrayList();
+    for (String interval : intervals.split(",")) {
+      intervalList.add(Interval.parse(interval.trim()));
+    }
```

Comment:
i guess we need to have something like this https://github.com/druid-io/druid/blob/0d745ee1207df4990ab92082ec33c75f5777757e/server/src/main/java/io/druid/server/http/DatasourcesResource.java#L346